packaging/CPackOptions.cmake

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019-2019 the openage authors. See copying.md for legal info.
+
+# Keeps all required CPack configuration for the Generators in one place and
+# CPack calls this file for every generator, keep this in mind, if you use
+# variables or functions outside of the conditional branch
+
+
+########################################################################
+# Continuous Integration
+
+# Filename addition for architecture, set from environment variables
+set(CPACK_PACKAGE_ARCHITECTURE "-$ENV{TARGET_PLATFORM}")
+
+# Filename addition for nightly-builds, set from environment variables
+if(DEFINED ENV{IS_NIGHTLY})
+    set(PACKAGE_NIGHTLY "_NIGHTLY")
+else()
+    set(PACKAGE_NIGHTLY "")
+endif()
+
+
+if(CPACK_GENERATOR MATCHES "NSIS")
+    # Explanations to the following commands you can find here:
+    # https://cmake.org/cmake/help/latest/cpack_gen/nsis.html#cpack_gen:CPack%20NSIS%20Generator
+
+    ########################################################################
+    # Filename
+
+     set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-v${CPACK_PACKAGE_VERSION}${CPACK_PACKAGE_ARCHITECTURE}-installer${PACKAGE_NIGHTLY}")
```

Comment:
where is `CPACK_PACKAGE_VERSION` set?

Reply:
CMAKE sets that automatically with the project-command we discussed earlier. This command called in the top-level cmakelists.txt sets the version project-wide. CPack gets it automatically by this.